src/Shared/NativeMethodsShared.cs

```diff
@@ -460,6 +473,48 @@ public SystemInformationData()
         /// </remarks>
         internal static int MAX_PATH = 260;
 
+        /// <summary>
+        /// Gets an enum for the max path restrictions of the current OS.
+        /// </summary>
+        internal static MaxPathLimits OSMaxPathLimit
```

Comment:
Does this being in `shared` mean that we'll actually do this registry call once per assembly? That seems suboptimal.

Reply:
I was wondering if this belonged here since the runtime OS checks seem to all be here. Is there a better spot to put this?